honeybee_vtk/data.py

```diff
@@ -0,0 +1,161 @@
+        data files match with the grid identifiers. For other than grid objects, it will
+        be checked that only one data file is provided and the length of the data file
+        matches the length of Polydata in the model.
+
+        Args:
+            model: A honeybee-vtk model object.
+
+        Returns:
+            A boolean value.
+        """
+
+        # if object name is "grid" check that the name of files match the grid names
+        if self.object_name == AcceptedValues.names.value[-1]:
+
+            assert len(model.sensor_grids.data) > 0, 'Sensor grids are not loaded on'\
+                ' this model. Reload them using grid options.'
+
+            grid_names = [grid.identifier for grid in model.sensor_grids.data]
+            file_names = [Path(path).name.split('.')[0] for path in self.file_paths]
+            if len(grid_names) != len(file_names) or grid_names != file_names:
+                raise ValueError(
+                    'The number of files and the file names must match the grid'
+                    ' identifiers in HBJSON.'
+                )
+            return True
+
+        # if object_name is other than grid check that length of data matches the length
+        # of data in the model for that object.
+        elif self.object_name in AcceptedValues.names.value[:-1]:
+
+            if len(self.file_paths) == 0:
+                raise ValueError(
+                    'File path not found in the config file.'
+                )
+            elif len(self.file_paths) > 1:
+                raise ValueError(
+                    'Only one file path needs to be provided in order to load data on'
+                    f' {self.object_name}. Multiple files are provided in the config'
+                    ' file.'
+                )
+
+            file = open(self.file_paths[0], "r")
+            nonempty_line_count = len(
+                [line.strip("\n") for line in file if line != "\n"]
+            )
+            file.close()
+
+            if nonempty_line_count != len(
+                    model.get_modeldataset_from_string(self.object_name).data):
+                raise ValueError(
+                    'The length of data in the file does not match the number of'
+                    f' {self.object_name} objects in the model.'
+                )
+            return True
+
+        else:
+            raise ValueError(
+                ' object_name must be from one of these'
+                f' {AcceptedValues.names.value}. Instead got {self.object_name}.'
+            )
+
+
+class DataConfig(BaseModel):
+    """Config for the whole data config file."""
+
+    data: Dict[str, Data] = Field(
+        description='A dictionary to introduce data that you would like to mount.'
+        ' The key must be any text and the value must be DataConfig.'
+    )
+
+    def check_data(self, model: Model) -> bool:
+        return all([val.cross_check_data(model) for val in self.data.values()])
```

Comment:
I'm not sure if you need this object at all. The data key is the only key here - why don't just rename what is called `Data` right now to `DataConfig` or simple `Config` and use that instead?

Reply:
I spent some time trying to think about this. May be I am missing something here, but how will I be able to parse the whole file if I don't have any one object capturing all other objects? How will the following command work if I don't have this object?

`json_obj = DataConfig.parse_file(path)`

Reply:
You will not be able to but you don't have to do it all in a single go. You can read the file first as a JSON Array and then run the validation for each object in the array.

Reply:
Complied.